modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc

```diff
@@ -5890,6 +5890,118 @@ Total database accesses: 5, total allocated memory: 1488
 
 ======
 
+[role=label--new-5.21]
+[[query-plan-stateful-shortest-path-into]]
+== StatefulShortestPath(Into)
+// StatefulShortestPath(Into)
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE CONSTRAINT person_name_unique IF NOT EXISTS FOR (p:Person) REQUIRE (p.name) IS UNIQUE
+----
+////
+
+The `StatefulShortestPath(Into)` operator finds shortest paths between two previously matched node variables.
+It uses a bidirectional breadth-first search (BFS) algorithm, which performs two BFS invocations at the same time, one from the left boundary node and one from the right boundary node.
+Once a node is found by both BFS invocations, which indicates that it can be reached from both boundary nodes, the algorithm successfully terminates.
+When both BFS invocations terminate without any overlap in the nodes found, then there exists no valid path between both boundary nodes.
```

Comment:
```suggestion
As soon as one of the BFS invocations terminates and there is no overlap in the nodes found by both BFS invocations, then there exists no valid path between both boundary nodes.
```

Reply:
I will need to verify this first with the runtime-team

Reply:
Response from Alex A: "Alex G is currently on parental leave and he’s the team expert, but AFAIK yes, as soon as one side exhausts its search the other should terminate too."